lib/query_components/filter.rb

```diff
@@ -1,23 +1,27 @@
+require "query_components/user_filter"
+require "query_components/visibility_filter"
+require 'elasticsearch/queries'
+
 module QueryComponents
   class Filter < BaseComponent
+    include Elasticsearch::Queries
+
     def payload(excluding = [])
-      rejects = []
-      filters = []
+      user_filter = QueryComponents::UserFilter.new(search_params)
+      visibility_filter = QueryComponents::VisibilityFilter.new(search_params)
 
-      param_filters = search_params.filters.reject do |filter|
-        excluding.include?(filter.field_name)
-      end
+      filters = combine_filters(user_filter.filters_payload(excluding), :and)
```

Comment:
@rboulton yep it is a bit confusing, might add some comments to clarify that.

Your assumption is right, I know visibility is just for restricting stuff so I didn't bother implementing an empty filter payload for that one.

I'm not sure what you are getting at in your example though. Do you just want to make the filter/reject code more similar? I could do something like what you wrote with the filter_components array just containing the one item, which would be clearer than adding extra payload methods I think...?